frontend/src/components/ResponsiveTextArea.jsx

```diff
@@ -0,0 +1,37 @@
+// NPM Packages
+import React, { useRef, useEffect} from 'react';
+
+export default function AutoFitContentPlaceholder({
+  placeholder,
+  contentText,
+  onFormContentChange,
+  maxLength
+}) {
+  // Local State
+  const textArea = useRef(null);
+
+  // Method
+  // Make textArea placeholder's height adaptive to the content
+  useEffect(() => {
+    const originalHeight = '52px';
+    textArea.current.style.height = originalHeight;
+    // counter 4px of padding included in scrollHeight
+    const newHeight = textArea.current.scrollHeight - 4;
+    textArea.current.style.height = `${newHeight}px`;
+    if (newHeight > 200) {
+      textArea.current.style.overflowY = 'scroll';
+    } else {
+      textArea.current.style.overflowY = 'none';
+    }
+  }, [contentText]);
+
+  return (
+    <textarea
+      ref={textArea}
+      className="form-input"
+      placeholder={placeholder}
+      value={contentText}
+      onChange={(e) => onFormContentChange(e.target.value)}
+     />
```

Comment:
do we need to add maxLength={maxLength}?

Reply:
Yes